ui/hooks/useNotificationTimeouts.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { INotification } from '@metamask/notification-services-controller/notification-services';
+import { deleteNotificationsById } from '../store/actions';
+import { NOTIFICATIONS_EXPIRATION_DELAY } from '../helpers/constants/notifications';
+
+/**
+ * This hook is used to enforce lifecycles for snap notifications.
+ * Upon dismount, the hook will clear timeouts for notifications
+ * that don't exist anymore.
+ *
+ * @param notifications - The current list of notifications
+ * @returns A function that creates a timeout to delete a notification
+ * and stores the timeout id.
+ */
+export const useSnapNotificationTimeouts = (notifications: INotification[]) => {
```

Comment:
This is being used for snap notifications right now but can be expanded to handle any other `locallyPersistedNotificationTypes` in the future: https://github.com/MetaMask/core/pull/4809/files#diff-df38af79c369a79900ff0c263ce48f4a7a7d8ef1ec2b350eaef11f93e66c5695R177

Reply:
See https://github.com/MetaMask/metamask-extension/pull/27975#discussion_r1809611582